[libqglviewer] Build error on x64-windows

Package: libqglviewer:x64-windows@2.9.1#1

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.29.30152.0
-    vcpkg-tool version: 2024-01-11-710a3116bbd615864eef5f9010af178034cb9b44
    vcpkg-scripts version: unknown

**To Reproduce**

`vcpkg install libQGLviewer:x64-windows`
**Failure logs**

```
-- Using cached GillesDebunne-libQGLViewer-v2.9.1.tar.gz.
-- Extracting source G:/vcpkg-2024.01.12/downloads/GillesDebunne-libQGLViewer-v2.9.1.tar.gz
-- Applying patch Add-compile-definitions.patch
-- Using source at G:/vcpkg-2024.01.12/buildtrees/libqglviewer/src/v2.9.1-22b59475ad.clean
-- Found external ninja('1.10.2').
-- Configuring x64-windows
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v -j7
    Working Directory: G:/vcpkg-2024.01.12/buildtrees/libqglviewer/x64-windows-dbg
    See logs for more information:
      G:\vcpkg-2024.01.12\buildtrees\libqglviewer\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/libqglviewer/portfile.cmake:14 (vcpkg_cmake_install)
  scripts/ports.cmake:170 (include)



```
<details><summary>G:\vcpkg-2024.01.12\buildtrees\libqglviewer\install-x64-windows-dbg-out.log</summary>

```
Change Dir: 'G:/vcpkg-2024.01.12/buildtrees/libqglviewer/x64-windows-dbg'

Run Build Command(s): "H:/Visual Studio 2019/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -v -v -j7 install
[1/156] C:\Windows\system32\cmd.exe /C "cd /D G:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg && "C:\Program Files\CMake\bin\cmake.exe" -E cmake_autogen G:/vcpkg-2024.01.12/buildtrees/libqglviewer/x64-windows-dbg/CMakeFiles/QGLViewer_autogen.dir/AutogenInfo.json Debug && "C:\Program Files\CMake\bin\cmake.exe" -E touch G:/vcpkg-2024.01.12/buildtrees/libqglviewer/x64-windows-dbg/QGLViewer_autogen/timestamp && "C:\Program Files\CMake\bin\cmake.exe" -E cmake_transform_depfile Ninja gccdepfile G:/vcpkg-2024.01.12/buildtrees/libqglviewer/src/v2.9.1-22b59475ad.clean G:/vcpkg-2024.01.12/buildtrees/libqglviewer/src/v2.9.1-22b59475ad.clean G:/vcpkg-2024.01.12/buildtrees/libqglviewer/x64-windows-dbg G:/vcpkg-2024.01.12/buildtrees/libqglviewer/x64-windows-dbg G:/vcpkg-2024.01.12/buildtrees/libqglviewer/x64-windows-dbg/QGLViewer_autogen/deps G:/vcpkg-2024.01.12/buildtrees/libqglviewer/x64-windows-dbg/CMakeFiles/d/3e518cf14934bedca6cb9af1f7b4ae31c5c0fd075b0503133cc431a64903c62f.d"
[2/156] H:\VISUAL~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe   /TP -DCREATE_QGLVIEWER_DLL -DNOMINMAX -DQGLViewer_EXPORTS -DQT_CORE_LIB -DQT_GUI_LIB -DQT_OPENGLWIDGETS_LIB -DQT_OPENGL_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -DUNICODE -DWIN32 -DWIN64 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -D_UNICODE -D_WIN64 -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg\QGLViewer_autogen\include -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtCore -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6 -external:IG:\vcpkg-2024.01.12\installed\x64-windows\share\Qt6\mkspecs\win32-msvc -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtWidgets -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtGui -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtXml -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtOpenGL -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtOpenGLWidgets -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -std:c++17 -Zc:__cplusplus -permissive- -utf-8 /showIncludes /FoCMakeFiles\QGLViewer.dir\QGLViewer\VRender\gpc.cpp.obj /FdCMakeFiles\QGLViewer.dir\ /FS -c G:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean\QGLViewer\VRender\gpc.cpp
[3/156] H:\VISUAL~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe   /TP -DCREATE_QGLVIEWER_DLL -DNOMINMAX -DQGLViewer_EXPORTS -DQT_CORE_LIB -DQT_GUI_LIB -DQT_OPENGLWIDGETS_LIB -DQT_OPENGL_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -DUNICODE -DWIN32 -DWIN64 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -D_UNICODE -D_WIN64 -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg\QGLViewer_autogen\include -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtCore -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6 -external:IG:\vcpkg-2024.01.12\installed\x64-windows\share\Qt6\mkspecs\win32-msvc -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtWidgets -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtGui -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtXml -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtOpenGL -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtOpenGLWidgets -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -std:c++17 -Zc:__cplusplus -permissive- -utf-8 /showIncludes /FoCMakeFiles\QGLViewer.dir\QGLViewer\VRender\Primitive.cpp.obj /FdCMakeFiles\QGLViewer.dir\ /FS -c G:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean\QGLViewer\VRender\Primitive.cpp
FAILED: CMakeFiles/QGLViewer.dir/QGLViewer/VRender/Primitive.cpp.obj 
H:\VISUAL~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe   /TP -DCREATE_QGLVIEWER_DLL -DNOMINMAX -DQGLViewer_EXPORTS -DQT_CORE_LIB -DQT_GUI_LIB -DQT_OPENGLWIDGETS_LIB -DQT_OPENGL_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -DUNICODE -DWIN32 -DWIN64 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -D_UNICODE -D_WIN64 -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg\QGLViewer_autogen\include -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtCore -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6 -external:IG:\vcpkg-2024.01.12\installed\x64-windows\share\Qt6\mkspecs\win32-msvc -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtWidgets -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtGui -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtXml -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtOpenGL -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtOpenGLWidgets -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -std:c++17 -Zc:__cplusplus -permissive- -utf-8 /showIncludes /FoCMakeFiles\QGLViewer.dir\QGLViewer\VRender\Primitive.cpp.obj /FdCMakeFiles\QGLViewer.dir\ /FS -c G:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean\QGLViewer\VRender\Primitive.cpp
G:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean\QGLViewer\VRender\Primitive.cpp(121): error C2039: '<<': is not a member of 'vrender'
G:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean\QGLViewer\VRender\Primitive.h(20): note: see declaration of 'vrender'
[4/156] H:\VISUAL~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe   /TP -DCREATE_QGLVIEWER_DLL -DNOMINMAX -DQGLViewer_EXPORTS -DQT_CORE_LIB -DQT_GUI_LIB -DQT_OPENGLWIDGETS_LIB -DQT_OPENGL_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -DUNICODE -DWIN32 -DWIN64 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -D_UNICODE -D_WIN64 -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg\QGLViewer_autogen\include -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtCore -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6 -external:IG:\vcpkg-2024.01.12\installed\x64-windows\share\Qt6\mkspecs\win32-msvc -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtWidgets -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtGui -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtXml -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtOpenGL -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtOpenGLWidgets -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -std:c++17 -Zc:__cplusplus -permissive- -utf-8 /showIncludes /FoCMakeFiles\QGLViewer.dir\QGLViewer\VRender\BSPSortMethod.cpp.obj /FdCMakeFiles\QGLViewer.dir\ /FS -c G:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean\QGLViewer\VRender\BSPSortMethod.cpp
G:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean\QGLViewer\VRender\BSPSortMethod.cpp(40): warning C4267: 'initializing': conversion from 'size_t' to 'unsigned int', possible loss of data
G:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean\QGLViewer\VRender\BSPSortMethod.cpp(272): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
G:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean\QGLViewer\VRender\BSPSortMethod.cpp(555): warning C4267: 'initializing': conversion from 'size_t' to 'unsigned int', possible loss of data
G:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean\QGLViewer\VRender\BSPSortMethod.cpp(561): warning C4267: '=': conversion from 'size_t' to 'unsigned int', possible loss of data
[5/156] H:\VISUAL~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe   /TP -DCREATE_QGLVIEWER_DLL -DNOMINMAX -DQGLViewer_EXPORTS -DQT_CORE_LIB -DQT_GUI_LIB -DQT_OPENGLWIDGETS_LIB -DQT_OPENGL_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -DUNICODE -DWIN32 -DWIN64 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -D_UNICODE -D_WIN64 -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg\QGLViewer_autogen\include -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtCore -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6 -external:IG:\vcpkg-2024.01.12\installed\x64-windows\share\Qt6\mkspecs\win32-msvc -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtWidgets -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtGui -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtXml -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtOpenGL -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtOpenGLWidgets -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -std:c++17 -Zc:__cplusplus -permissive- -utf-8 /showIncludes /FoCMakeFiles\QGLViewer.dir\QGLViewer\VRender\BackFaceCullingOptimizer.cpp.obj /FdCMakeFiles\QGLViewer.dir\ /FS -c G:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean\QGLViewer\VRender\BackFaceCullingOptimizer.cpp
[6/156] H:\VISUAL~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe   /TP -DCREATE_QGLVIEWER_DLL -DNOMINMAX -DQGLViewer_EXPORTS -DQT_CORE_LIB -DQT_GUI_LIB -DQT_OPENGLWIDGETS_LIB -DQT_OPENGL_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -DUNICODE -DWIN32 -DWIN64 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -D_UNICODE -D_WIN64 -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg\QGLViewer_autogen\include -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtCore -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6 -external:IG:\vcpkg-2024.01.12\installed\x64-windows\share\Qt6\mkspecs\win32-msvc -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtWidgets -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtGui -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtXml -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtOpenGL -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtOpenGLWidgets -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -std:c++17 -Zc:__cplusplus -permissive- -utf-8 /showIncludes /FoCMakeFiles\QGLViewer.dir\QGLViewer\VRender\FIGExporter.cpp.obj /FdCMakeFiles\QGLViewer.dir\ /FS -c G:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean\QGLViewer\VRender\FIGExporter.cpp
G:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean\QGLViewer\VRender\FIGExporter.cpp(84): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data
[7/156] H:\VISUAL~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe   /TP -DCREATE_QGLVIEWER_DLL -DNOMINMAX -DQGLViewer_EXPORTS -DQT_CORE_LIB -DQT_GUI_LIB -DQT_OPENGLWIDGETS_LIB -DQT_OPENGL_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -DUNICODE -DWIN32 -DWIN64 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -D_UNICODE -D_WIN64 -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg\QGLViewer_autogen\include -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtCore -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6 -external:IG:\vcpkg-2024.01.12\installed\x64-windows\share\Qt6\mkspecs\win32-msvc -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtWidgets -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtGui -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtXml -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtOpenGL -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtOpenGLWidgets -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -std:c++17 -Zc:__cplusplus -permissive- -utf-8 /showIncludes /FoCMakeFiles\QGLViewer.dir\QGLViewer\VRender\EPSExporter.cpp.obj /FdCMakeFiles\QGLViewer.dir\ /FS -c G:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean\QGLViewer\VRender\EPSExporter.cpp
G:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean\QGLViewer\VRender\EPSExporter.cpp(114): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data
[8/156] H:\VISUAL~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe   /TP -DCREATE_QGLVIEWER_DLL -DNOMINMAX -DQGLViewer_EXPORTS -DQT_CORE_LIB -DQT_GUI_LIB -DQT_OPENGLWIDGETS_LIB -DQT_OPENGL_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -DUNICODE -DWIN32 -DWIN64 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -D_UNICODE -D_WIN64 -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg\QGLViewer_autogen\include -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtCore -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6 -external:IG:\vcpkg-2024.01.12\installed\x64-windows\share\Qt6\mkspecs\win32-msvc -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtWidgets -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtGui -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtXml -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtOpenGL -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtOpenGLWidgets -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -std:c++17 -Zc:__cplusplus -permissive- -utf-8 /showIncludes /FoCMakeFiles\QGLViewer.dir\QGLViewer\VRender\Exporter.cpp.obj /FdCMakeFiles\QGLViewer.dir\ /FS -c G:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean\QGLViewer\VRender\Exporter.cpp
G:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean\QGLViewer\VRender\Exporter.cpp(32): warning C4267: 'initializing': conversion from 'size_t' to 'unsigned int', possible loss of data
[9/156] H:\VISUAL~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe   /TP -DCREATE_QGLVIEWER_DLL -DNOMINMAX -DQGLViewer_EXPORTS -DQT_CORE_LIB -DQT_GUI_LIB -DQT_OPENGLWIDGETS_LIB -DQT_OPENGL_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -DUNICODE -DWIN32 -DWIN64 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -D_UNICODE -D_WIN64 -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\src\v2.9.1-22b59475ad.clean -IG:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg\QGLViewer_autogen\include -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtCore -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6 -external:IG:\vcpkg-2024.01.12\installed\x64-windows\share\Qt6\mkspecs\win32-msvc -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtWidgets -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtGui -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtXml -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtOpenGL -external:IG:\vcpkg-2024.01.12\installed\x64-windows\include\Qt6\QtOpenGLWidgets -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -std:c++17 -Zc:__cplusplus -permissive- -utf-8 /showIncludes /FoCMakeFiles\QGLViewer.dir\QGLViewer_autogen\mocs_compilation.cpp.obj /FdCMakeFiles\QGLViewer.dir\ /FS -c G:\vcpkg-2024.01.12\buildtrees\libqglviewer\x64-windows-dbg\QGLViewer_autogen\mocs_compilation.cpp
ninja: build stopped: subcommand failed.
```
</details>

--- a/ports/libqglviewer/Fix-error-c2039.patch
+++ b/ports/libqglviewer/Fix-error-c2039.patch
@@ -1,0 +1,52 @@
+diff --git a/QGLViewer/VRender/NVector3.h b/QGLViewer/VRender/NVector3.h
+index 40b7f98..f2d8305 100644
+--- a/QGLViewer/VRender/NVector3.h
++++ b/QGLViewer/VRender/NVector3.h
+@@ -7,6 +7,8 @@
+ namespace vrender
+ {
+   class Vector3;
++  class NVector3;
++  std::ostream& operator<<(std::ostream &out,const NVector3 &u);
+ 
+   class NVector3
+   {
+diff --git a/QGLViewer/VRender/Primitive.h b/QGLViewer/VRender/Primitive.h
+index 88ab11d..d38470d 100644
+--- a/QGLViewer/VRender/Primitive.h
++++ b/QGLViewer/VRender/Primitive.h
+@@ -21,6 +21,7 @@ namespace vrender
+ {
+ 	class Feedback3DColor ;
+ 	class Primitive ;
++    std::ostream& operator<<(std::ostream&, const Feedback3DColor&) ;
+ 
+ #define EPS_SMOOTH_LINE_FACTOR 0.06  /* Lower for better smooth lines. */
+ 
+diff --git a/QGLViewer/VRender/Vector2.h b/QGLViewer/VRender/Vector2.h
+index f6aaaf3..7b9b82a 100644
+--- a/QGLViewer/VRender/Vector2.h
++++ b/QGLViewer/VRender/Vector2.h
+@@ -6,7 +6,9 @@
+ 
+ namespace vrender
+ {
++  class Vector2;
+   class Vector3;
++  std::ostream& operator<< (std::ostream&,const Vector2&);
+ 
+   class Vector2
+ 	{
+diff --git a/QGLViewer/VRender/Vector3.h b/QGLViewer/VRender/Vector3.h
+index 32597e8..f6d5099 100644
+--- a/QGLViewer/VRender/Vector3.h
++++ b/QGLViewer/VRender/Vector3.h
+@@ -10,6 +10,8 @@
+ namespace vrender
+ {
+   class NVector3;
++  class Vector3;
++  std::ostream& operator<< (std::ostream&, const Vector3&);
+ 
+ 	class Vector3
+ 	{

--- a/ports/libqglviewer/portfile.cmake
+++ b/ports/libqglviewer/portfile.cmake
@@ -1,10 +1,11 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO GillesDebunne/libQGLViewer
-    REF v2.9.1
+    REF "v${VERSION}"
     SHA512 09bfc5c0f07e51625a9af0094b83f40f84ead55a67c6e492c9702521f58c6b461bc840382fb73b64d16ad71a0a2a75d04aa12a77a78ced0a19e0e784e8d36bd7
     PATCHES
         Add-compile-definitions.patch
+        Fix-error-c2039.patch #https://github.com/GillesDebunne/libQGLViewer/pull/80
 )
 
 vcpkg_cmake_configure(
@@ -17,4 +18,4 @@ vcpkg_fixup_pkgconfig()
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 
-file(INSTALL ${SOURCE_PATH}/LICENCE DESTINATION ${CURRENT_PACKAGES_DIR}/share/libqglviewer RENAME copyright)
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENCE")

--- a/ports/libqglviewer/vcpkg.json
+++ b/ports/libqglviewer/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "libqglviewer",
   "version": "2.9.1",
-  "port-version": 1,
+  "port-version": 2,
   "description": "libQGLViewer is an open source C++ library based on Qt that eases the creation of OpenGL 3D viewers.",
   "homepage": "http://libqglviewer.com/",
   "license": "GPL-2.0-or-later",
@@ -13,10 +13,6 @@
     },
     {
       "name": "vcpkg-cmake",
-      "host": true
-    },
-    {
-      "name": "vcpkg-cmake-config",
       "host": true
     }
   ]

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4742,7 +4742,7 @@
     },
     "libqglviewer": {
       "baseline": "2.9.1",
-      "port-version": 1
+      "port-version": 2
     },
     "libqrencode": {
       "baseline": "4.1.1",

--- a/versions/l-/libqglviewer.json
+++ b/versions/l-/libqglviewer.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "bfaae9deb95ec0d71a5d9ba8342fd3c83c8871fa",
+      "version": "2.9.1",
+      "port-version": 2
+    },
+    {
       "git-tree": "fb8cb610ce82055671ffb7e84836da4a02cc3127",
       "version": "2.9.1",
       "port-version": 1

